api/v1alpha1/perconaservermysql_types.go

```diff
@@ -128,6 +129,79 @@ type PMMSpec struct {
+	ImagePullPolicy          corev1.PullPolicy             `json:"imagePullPolicy,omitempty"`
+	ServiceAccountName       string                        `json:"serviceAccountName,omitempty"`
+	ContainerSecurityContext *corev1.SecurityContext       `json:"containerSecurityContext,omitempty"`
+	Resources                corev1.ResourceRequirements   `json:"resources,omitempty"`
+	Storages                 map[string]*BackupStorageSpec `json:"storages,omitempty"`
+}
+
+type BackupStorageType string
+
+const (
+	BackupStorageFilesystem BackupStorageType = "filesystem"
+	BackupStorageS3         BackupStorageType = "s3"
+	BackupStorageGCS        BackupStorageType = "gcs"
+	BackupStorageAzure      BackupStorageType = "azure"
+)
+
+type BackupStorageSpec struct {
+	Type                     BackupStorageType           `json:"type"`
+	Volume                   *VolumeSpec                 `json:"volumeSpec,omitempty"`
+	S3                       *BackupStorageS3Spec        `json:"s3,omitempty"`
+	GCS                      *BackupStorageGCSSpec       `json:"gcs,omitempty"`
+	Azure                    *BackupStorageAzureSpec     `json:"azure,omitempty"`
+	NodeSelector             map[string]string           `json:"nodeSelector,omitempty"`
+	Resources                corev1.ResourceRequirements `json:"resources,omitempty"`
+	Affinity                 *corev1.Affinity            `json:"affinity,omitempty"`
+	Tolerations              []corev1.Toleration         `json:"tolerations,omitempty"`
+	Annotations              map[string]string           `json:"annotations,omitempty"`
+	Labels                   map[string]string           `json:"labels,omitempty"`
+	SchedulerName            string                      `json:"schedulerName,omitempty"`
+	PriorityClassName        string                      `json:"priorityClassName,omitempty"`
+	PodSecurityContext       *corev1.PodSecurityContext  `json:"podSecurityContext,omitempty"`
+	ContainerSecurityContext *corev1.SecurityContext     `json:"containerSecurityContext,omitempty"`
+	RuntimeClassName         *string                     `json:"runtimeClassName,omitempty"`
+	VerifyTLS                *bool                       `json:"verifyTLS,omitempty"`
+}
+
+type BackupStorageS3Spec struct {
+	Bucket            string `json:"bucket"`
+	CredentialsSecret string `json:"credentialsSecret"`
+	Region            string `json:"region,omitempty"`
+	EndpointURL       string `json:"endpointUrl,omitempty"`
+	StorageClass      string `json:"storageClass,omitempty"`
+}
+
+type BackupStorageGCSSpec struct {
+	Bucket            string `json:"bucket"`
+	CredentialsSecret string `json:"credentialsSecret"`
+	EndpointURL       string `json:"endpointUrl,omitempty"`
+
+	// STANDARD, NEARLINE, COLDLINE, ARCHIVE
+	StorageClass string `json:"storageClass,omitempty"`
+}
+
+type BackupStorageAzureSpec struct {
+	// An Azure storage account is a unique namespace to access and store your Azure data objects.
+	AccountName string `json:"accountName"`
```

Comment:
Should we remove this here if we have added `AZURE_STORAGE_ACCOUNT_NAME` in the secret?